xrtl/tools/target_config/target_config.cc

```diff
@@ -14,6 +14,6 @@
 
 #include "xrtl/tools/target_config/target_config.h"
 
-namespace {
-void Dummy() {}  // Quiet linker.
-}  // namespace
+#if defined(__clang__) || defined(__GNUC__)
+static int dummy __attribute__((unused, used)) = 0;
```

Comment:
These can't pull in macros.h (as that'd create a cycle), so no unfortunately :( I'll add a comment to point back at the macro in case someone is wondering what this is.